distributed/scheduler.py

```diff
@@ -1213,7 +1213,11 @@ def get_worker_service_addr(self, worker, service_name, protocol=False):
             self.status = "running"
             logger.info("  Scheduler at: %25s", self.address)
             for k, v in self.services.items():
-                logger.info("%11s at: %25s", k, "%s:%d" % (listen_ip, v.port))
+                if hasattr(v, "server"):
+                    ip = v.server.address
```

Comment:
A few questions:

When does this come up?  Presumably for the BokehServer stuff?

What if this object has a `server` attribute but not an `address` attribute?  

Does address typically mean just the IP or the full ip/port pair?